### PR TITLE
Clean up TODO items in tests

### PR DESCRIPTION
**Pull request by astralcai**
- Removed TODO items which reference issues already resolved.
- Removed tests for features no longer supported (nested tapes)
- Updated many tests to use the now available qml.assert_equal
- Uncommented tests and test cases that were commented out because of an issue that is already resolved (also a reminder to use xfail to mark tests that are expected to fail due to some unresolved issue, instead of commenting it out)
Other minor changes.
- Removed some xfails (if an xfail is called inside the test instead of added as a marker, we actually do not get an XPASS when the test could've passed without the xfail)
- Removed some skips (initially skipped because of some feature that were unsupported at the time but supported now)
- Added references to shortcut stories in some TODO comments.

[sc-76863]

### CODE REVIEW

File: tests/gradients/parameter_shift/test_parameter_shift.py [hunk 1]
```diff
@@ -2570,7 +2571,7 @@ def test_output_shape_matches_qnode_probs(self, cost, exp_shape, exp_type):
         assert np.allclose(np.squeeze(np.array(res_parshift)).shape, exp_shape)
 
     # TODO: revisit the following test when the Autograd interface supports
-    # parameter-shift with the new return type system
+    #       parameter-shift with the new return type system
```
Comment (albi3ro) [inline tests/gradients/parameter_shift/test_parameter_shift.py:2574]: What in particular are we needing to revist? Given we fully ported to the new return type system two years ago

Comment (albi3ro) [reply tests/gradients/parameter_shift/test_parameter_shift.py:2574]: We may also want to rewrite this test to use the new device interface at some point.

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 2]
```diff
@@ -2308,9 +2290,7 @@ def test_tf(self, broadcast, tol):
         # assert np.allclose(hess[0][:, 2:5], np.zeros([2, 3, 3]), atol=tol, rtol=0)
         # assert np.allclose(hess[1][:, -1], np.zeros([2, 1, 1]), atol=tol, rtol=0)
 
-    # TODO: Torch support for param-shift
     @pytest.mark.torch
-    @pytest.mark.xfail
```
Comment (albi3ro) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py:2313]: how was this still failing?

File: tests/gradients/parameter_shift/test_parameter_shift.py (hunk 1 above)
Comment (astralcai) [reply tests/gradients/parameter_shift/test_parameter_shift.py:2574]: > What in particular are we needing to revisit?

I don't know 😅 but yeah we definitely want to rewrite many of the gradient tests in general at some point.

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py (hunk 2 above)
Comment (astralcai) [reply tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py:2313]: It was failing because `res` was a tuple and a tuple didn't have a `detach` method, see update made below.

**Review by albi3ro — APPROVED** [review]

File: tests/gradients/parameter_shift/test_parameter_shift.py [hunk 3]
```diff
@@ -740,25 +741,25 @@ def test_all_zero_diff_methods_multiple_returns_tape(self):
...
+            qml.Rot(*params, wires=0)
+            return qml.probs([2, 3])
 
-    #     params = np.array([0.5, 0.5, 0.5], requires_grad=True)
+        params = np.array([0.5, 0.5, 0.5], requires_grad=True)
 
-    #     result = qml.gradients.param_shift(circuit)(params)
-    #     assert np.allclose(result, np.zeros((4, 3)), atol=0, rtol=0)
+        result = qml.gradients.param_shift(circuit)(params)
+        assert np.allclose(result, np.zeros((4, 3)), atol=0, rtol=0)
```
[the remark this answers is not in the document]
Comment (astralcai) [reply tests/gradients/parameter_shift/test_parameter_shift.py:758]: > Still wonder why set zero atol in a numpy numeric program.

🤷🏻

**Review by JerryChen97 — APPROVED** [review]

**Comment by JerryChen97** [comment]
I saw a fail previously though. Don't know if we already fixed it: https://github.com/PennyLaneAI/pennylane/actions/runs/12262931706/job/34213396999?pr=6690
`tests/test_compiler.py::TestCatalystMCMs::test_dynamic_one_shot_simple`

**Comment by JerryChen97** [comment]
It seems one file to black again and one file to remove redundant import of `QuantumTape` and we are ready to see codecov!

**Comment by astralcai** [comment]
> I saw a fail previously though. Don't know if we already fixed it: https://github.com/PennyLaneAI/pennylane/actions/runs/12262931706/job/34213396999?pr=6690 `tests/test_compiler.py::TestCatalystMCMs::test_dynamic_one_shot_simple`

The test was failing due to a bug, now xfailed for the proper reason: https://github.com/PennyLaneAI/pennylane/issues/6700

File: tests/test_compiler.py [hunk 4]
```diff
@@ -870,23 +870,29 @@ def circuit(x):
 class TestCatalystMCMs:
     """Test dynamic_one_shot with Catalyst."""
 
-    @pytest.mark.xfail(reason="requires simultaneous catalyst pr")
-    @pytest.mark.parametrize("measure_f", [qml.counts, qml.expval, qml.probs])
+    @pytest.mark.parametrize(
+        "measure_f",
+        [
+            # https://github.com/PennyLaneAI/pennylane/issues/6700
+            pytest.param(qml.count, marks=pytest.mark.xfail),
```
Comment (JerryChen97) [inline tests/test_compiler.py:877]:
```suggestion
            pytest.param(qml.counts, marks=pytest.mark.xfail),
```

Comment (astralcai) [reply tests/test_compiler.py]: Ooooops
